.github/workflows/linting.yml

```diff
@@ -0,0 +1,41 @@
+# Copyright 2022 GlobalFoundries PDK Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Code Linting
+on:
+  pull_request:
+  push:
+
+jobs:
+  test_code:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      max-parallel: 12
+      matrix:
+        python-version: ["3.10"]
+        os: [ubuntu-latest]
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: "pip"
+      - name: Install dependencies
+        run: |
+          pip install flake8 pytest pytest_regressions
```

Comment:
Should these be in a `requirements.txt` or `requirements.dev.txt` file? Then we can just `pip install -r requirements.txt`?

Reply:
+1

Reply:
@mithro and @proppy I could add that `requirements.txt`. But I wasn't sure what to do here as those only are used as part of the github actions only. I keep all the really required packages in `requirements.txt` file.

What's your recommendation here add it to `requirments.txt` or not? Keep in mind that those are only used for qualification purposes.

Reply:
What to put in your `requirements.txt` is confusing even for long time Python users.

My general guidance is that;
 * `requirements.txt` should be anything needed to setup the Python environment to run the tests successfully.
 * `requirements.dev.txt` should be used to install anything needed to do development. 

You can even include `requirements.txt` inside `requirements.dev.txt` with a `-r requirements.txt` as the first line in `requirements.txt` as shown at https://realpython.com/lessons/production-vs-development-dependencies/

If you are creating a Python package then there is also the `install_requires` in the `setup.py` file which describes what is need to run your Python package. There is some older guidance about this topic at https://caremad.io/posts/2013/07/setup-vs-requirement/ and some newer guidance at https://packaging.python.org/en/latest/discussions/install-requires-vs-requirements/

(Just FYI the full `requirements.txt` format is described at https://pip.pypa.io/en/latest/reference/requirements-file-format/#requirements-file-format)

If you have binary dependencies that are not available on PyPi then you need to use conda and an `environment.yml` file (see https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#create-env-file-manually), An `environment.yml` files is like a `requirements.txt` on steroids but also lets to specify conda packages. To make setting up a conda environment quick and easy, I created the `make-env` tool which can be found at https://github.com/SymbiFlow/make-env which will do all the work to download conda and setup the Python environment.

Reply:
@mithro and @proppy Same as above: 
https://github.com/efabless/globalfoundries-pdk-libs-gf180mcu_fd_pv/commit/19cb6f2ac68e59fd009b713a170bceee44487daa